pkg/cache/internal/informers_map.go

```diff
@@ -409,41 +409,15 @@ func createMetadataListWatch(gvk schema.GroupVersionKind, ip *specificInformersM
 	}, nil
 }
 
-type gvkFixupWatcher struct {
-	watcher watch.Interface
-	ch      chan watch.Event
-	gvk     schema.GroupVersionKind
-	wg      sync.WaitGroup
-}
-
 func newGVKFixupWatcher(gvk schema.GroupVersionKind, watcher watch.Interface) watch.Interface {
-	ch := make(chan watch.Event)
-	w := &gvkFixupWatcher{
-		gvk:     gvk,
-		watcher: watcher,
-		ch:      ch,
-	}
-	w.wg.Add(1)
-	go w.run()
-	return w
-}
-
-func (w *gvkFixupWatcher) run() {
-	for e := range w.watcher.ResultChan() {
-		e.Object.GetObjectKind().SetGroupVersionKind(w.gvk)
-		w.ch <- e
-	}
-	w.wg.Done()
-}
-
-func (w *gvkFixupWatcher) Stop() {
-	w.watcher.Stop()
-	w.wg.Wait()
-	close(w.ch)
-}
-
-func (w *gvkFixupWatcher) ResultChan() <-chan watch.Event {
-	return w.ch
+	return watch.Filter(
+		watcher,
+		func(in watch.Event) (out watch.Event, keep bool) {
+			in.DeepCopyInto(&out)
+			out.Object.GetObjectKind().SetGroupVersionKind(gvk)
```

Comment:
The original implementation mutated the watch event. 
I've used DeepCopy here, but I'm not sure if it's necessary.

Reply:
I don't _think_ DeepCopy is necessary.

Reply:
The `in watch.Event`  is passed in by value, but it has a pointer to a `runtime.Object` (e.g. `&metav1.PartialObjectMetadata` ) which *might* be mutated elsewhere, so I thought it was safest to DeepCopy.

Reply:
Yeah, no strong objections to DeepCopy from me!

Reply:
Don't think it is necessary, but no strong objections from me too :)

Reply:
+1 to remove deepcopy

Reply:
Ok. Done.